src/auto/file_attribute_matcher.rs

```diff
@@ -0,0 +1,70 @@
+// This file was generated by gir (https://github.com/gtk-rs/gir)
+// from gir-files (https://github.com/gtk-rs/gir-files)
+// DO NOT EDIT
+
+use gio_sys;
+use glib::GString;
+use glib::translate::*;
+use std::fmt;
+
+glib_wrapper! {
+    #[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
+    pub struct FileAttributeMatcher(Shared<gio_sys::GFileAttributeMatcher>);
+
+    match fn {
+        ref => |ptr| gio_sys::g_file_attribute_matcher_ref(ptr),
+        unref => |ptr| gio_sys::g_file_attribute_matcher_unref(ptr),
+        get_type => || gio_sys::g_file_attribute_matcher_get_type(),
+    }
+}
+
+impl FileAttributeMatcher {
+    pub fn new(attributes: &str) -> FileAttributeMatcher {
+        unsafe {
+            from_glib_full(gio_sys::g_file_attribute_matcher_new(attributes.to_glib_none().0))
+        }
+    }
+
+    pub fn enumerate_namespace(&self, ns: &str) -> bool {
+        unsafe {
+            from_glib(gio_sys::g_file_attribute_matcher_enumerate_namespace(self.to_glib_none().0, ns.to_glib_none().0))
+        }
+    }
+
+    pub fn enumerate_next(&self) -> Option<GString> {
+        unsafe {
+            from_glib_none(gio_sys::g_file_attribute_matcher_enumerate_next(self.to_glib_none().0))
+        }
+    }
+
+    pub fn matches(&self, attribute: &str) -> bool {
+        unsafe {
+            from_glib(gio_sys::g_file_attribute_matcher_matches(self.to_glib_none().0, attribute.to_glib_none().0))
+        }
+    }
+
+    pub fn matches_only(&self, attribute: &str) -> bool {
+        unsafe {
+            from_glib(gio_sys::g_file_attribute_matcher_matches_only(self.to_glib_none().0, attribute.to_glib_none().0))
+        }
+    }
+
+    pub fn subtract(&self, subtract: &FileAttributeMatcher) -> Option<FileAttributeMatcher> {
```

Comment:
I don't think this can return `None`

Reply:
I'll check.

Reply:
It seems like it can: https://gitlab.gnome.org/GNOME/glib/blob/master/gio/gfileinfo.c#L2388

Reply:
Indeed, and it's valid to pass `NULL` too for both parameters

Reply:
Ah! I need to make (at least) the second one nullable then.